cpu/gd32v/Makefile.include

```diff
@@ -1,8 +1,37 @@
 RAM_START_ADDR ?= 0x20000000
-RAM_LEN ?= 32K
-
 ROM_START_ADDR ?= 0x08000000
-ROM_LEN ?= 128K
+
+ifneq (,$(filter gd32vf103%bt6 gd32vf103%bu6,$(CPU_MODEL)))
+  RAM_LEN ?= 32K
+  ROM_LEN ?= 128K
+else ifneq (,$(filter gd32vf103%8t6 gd32vf103%8u6,$(CPU_MODEL)))
+  RAM_LEN ?= 20K
+  ROM_LEN ?= 64K
+else ifneq (,$(filter gd32vf103%6t6 gd32vf103%6u6,$(CPU_MODEL)))
+  RAM_LEN ?= 10K
+  ROM_LEN ?= 32K
+else ifneq (,$(filter gd32vf103%4t6 gd32vf103%4u6,$(CPU_MODEL)))
+  RAM_LEN ?= 6K
+  ROM_LEN ?= 16K
+else
+  $(error CPU model $(CPU_MODEL) not supported)
+endif
+
+FW_ROM_LEN ?= $(shell printf "0x%x" $$(($(ROM_LEN:%K=%*1024))))
+
+RIOTBOOT_HDR_LEN ?= 0x400
+ifneq (,$(filter usbus_dfu tinyusb_dfu,$(USEMODULE)))
+  RIOTBOOT_LEN ?= 0x4000
+else
+  RIOTBOOT_LEN ?= 0x1000
+endif
+
+NUM_SLOTS ?= 2
+
+SLOT0_LEN ?= $(shell printf "0x%x" $$((($(ROM_LEN:%K=%*1024)-$(RIOTBOOT_LEN)) / $(NUM_SLOTS))))
```

Comment:
Would it work to move this to `sys/riotboot/Makefile.include` (and drop it from `cpu/cortexm_common/Makefile.include`) so that when this is modified (as is done in #18608) only a single place has to be updated

Reply:
Should work.

Reply:
Yep, works a1b75aeb0ebfc2772cbbbdb1e8e0c0ad11149a62